module/extension/Director/tests/when_causing_multi.cpp

```diff
@@ -0,0 +1,120 @@
+#include <catch.hpp>
+#include <nuclear>
+
+#include "Director.hpp"
+#include "TestBase.hpp"
+
+// Anonymous namespace to avoid name collisions
+namespace {
+
+    struct SimpleTask {};
+
+    template <int i>
+    struct Subtask {};
+
+    template <int i>
+    struct Helper {};
+
+    template <int i>
+    struct Condition {
+        enum Value { BLOCK, ALLOW } value;
+        Condition(const Value& v) : value(v) {}
+        operator int() const {
+            return value;
+        }
+    };
+
+    std::vector<std::string> events;
+
+    class TestReactor : public TestBase<TestReactor> {
+    public:
+        explicit TestReactor(std::unique_ptr<NUClear::Environment> environment)
+            : TestBase<TestReactor>(std::move(environment)) {
+
+            on<Provide<SimpleTask>, Needs<Subtask<1>>, Needs<Subtask<2>>>().then([this] {
+                events.push_back("task executed");
+                emit<Task>(std::make_unique<Subtask<1>>());
+                emit<Task>(std::make_unique<Subtask<2>>());
+            });
+
+
+            on<Provide<Subtask<1>>, When<Condition<1>, std::equal_to, Condition<1>::ALLOW>>().then([this] {  //
+                events.push_back("subtask 1 executed");
+            });
+            on<Provide<Subtask<2>>, When<Condition<2>, std::equal_to, Condition<2>::ALLOW>>().then([this] {  //
+                events.push_back("subtask 1 executed");
+            });
+
+
+            on<Provide<Helper<1>>>().then([this] { events.push_back("helper 1 waiting"); });
```

Comment:
did you want to trick the formatter here too or is this one fine as is :stuck_out_tongue:

Reply:
(There's another one just below this for the second Helper too)

Reply:
I don't mind this one being a single line